org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -214,12 +215,21 @@ public void handleOpen(DidOpenTextDocumentParams params) {
 					// ignored
 				}
 			}
-			//Resources belonging to the default project can only report syntax errors, because the project classpath is incomplete
 			IProject project = unit.getResource().getProject();
-			boolean reportOnlySyntaxErrors = project.equals(JavaLanguageServerPlugin.getProjectsManager().getDefaultProject());
+			// Resources belonging to the default project can only report syntax errors, because the project classpath is incomplete
+			boolean isDefaultProject = project.equals(JavaLanguageServerPlugin.getProjectsManager().getDefaultProject());
+			// Resources that aren't on the classpath can only report syntax errors
+			IJavaProject javaProject = JavaCore.create(project);
+			boolean isOnClassPath = javaProject != null && javaProject.isOnClasspath(unit);
+			boolean reportOnlySyntaxErrors = isDefaultProject || !isOnClassPath;
```

Comment:
Is this local variable needed here? It does not seem to be used on multiple places.